src/software/backend/wifi_backend.cpp

```diff
@@ -0,0 +1,79 @@
+#include "software/backend/wifi_backend.h"
+
+#include "shared/constants.h"
+#include "software/constants.h"
+#include "software/parameter/dynamic_parameters.h"
+#include "software/proto/message_translation/protobuf_message_translation.h"
+#include "software/util/design_patterns/generic_factory.h"
+
+const std::string WifiBackend::name = "wifi";
+
+WifiBackend::WifiBackend()
+    : network_input(Util::Constants::SSL_VISION_DEFAULT_MULTICAST_ADDRESS,
+                    Util::Constants::SSL_VISION_MULTICAST_PORT,
+                    Util::Constants::SSL_GAMECONTROLLER_MULTICAST_ADDRESS,
+                    Util::Constants::SSL_GAMECONTROLLER_MULTICAST_PORT,
+                    boost::bind(&WifiBackend::receiveWorld, this, _1),
+                    Util::DynamicParameters->getAIControlConfig()->getRefboxConfig(),
+                    Util::DynamicParameters->getCameraConfig()),
+      io_service()
+{
+    std::string network_interface =
```

Comment:
we should have a way to inject runtime arguments w/ the new dp system, this is the best I could do with what we have

the interface defaults to the value in the yaml file, and then the channel can change on the callback

thoughts on how we can do this better?